File: baekjoon_all/01000+/boj_1018.py
# Solve 2022-05-26
# Update 2025-09-05

import sys

input = lambda: sys.stdin.readline().rstrip()

n, m = map(int, input().split())
board = [input() for _ in range(n)]
ans = 64

for i in range(n - 7):
    for j in range(m - 7):
        cnt = 0

        for x in range(i, i + 8):
            for y in range(j, j + 8):
                if board[x][y] == "BW"[(x + y) % 2]:
                    cnt += 1

        ans = min(ans, cnt, 64 - cnt)

print(ans)

File: baekjoon_all/01000+/boj_1018.cpp
// Solve 2023-10-14
// Update 2025-09-05

#include <bits/stdc++.h>

#define FASTIO ios_base::sync_with_stdio(false);cin.tie(NULL);
#define ALL(v) v.begin(),v.end()
#define UNIQUE(v) v.erase(unique(v.begin(),v.end()),v.end());
#define SETW(n, c) cout << setw(n) << setfill(c);
#define SETP(n) cout << fixed << setprecision(n);

using namespace std;
using ll = long long;
using uint = unsigned int;
using ull = unsigned long long;
using ld = long double;
using pii = pair<int, int>;
using pll = pair<ll, ll>;
const int INF = 0x3f3f3f3f;
const int MOD = 1000000007;

int main() {
    FASTIO;

    int n, m;
    cin >> n >> m;

    vector<string> board(n, "");

    for (string &row : board) {
        cin >> row;
    }

    int ans = 32;

    for (int i = 7; i < n; i++) {
        for (int j = 7; j < m; j++) {
            int cnt = 0;

            for (int x = i - 7; x <= i; x++) {
                for (int y = j - 7; y <= j; y++) {
                    if ((x + y) % 2 == 0) {
                        if (board[x][y] == 'B')  {
                            cnt++;
                        }
                    }
                    else {
                        if (board[x][y] == 'W')  {
                            cnt++;
                        }
                    }
                }
            }

            ans = min(ans, min(cnt, 64 - cnt));
        }
    }

    cout << ans << '\n';
    return 0;
}
